week7/Homework7-start/Homework7/Models/API/APIStore.swift

```diff
@@ -0,0 +1,99 @@
+/// Copyright (c) 2024 Kodeco Inc.
+/// 
+/// Permission is hereby granted, free of charge, to any person obtaining a copy
+/// of this software and associated documentation files (the "Software"), to deal
+/// in the Software without restriction, including without limitation the rights
+/// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+/// copies of the Software, and to permit persons to whom the Software is
+/// furnished to do so, subject to the following conditions:
+/// 
+/// The above copyright notice and this permission notice shall be included in
+/// all copies or substantial portions of the Software.
+/// 
+/// Notwithstanding the foregoing, you may not use, copy, modify, merge, publish,
+/// distribute, sublicense, create a derivative work, and/or sell copies of the
+/// Software in any work that is designed, intended, or marketed for pedagogical or
+/// instructional purposes related to programming, coding, application development,
+/// or information technology.  Permission for such use, copying, modification,
+/// merger, publication, distribution, sublicensing, creation of derivative works,
+/// or sale is expressly withheld.
+/// 
+/// This project and source code may use libraries or frameworks that are
+/// released under various Open-Source licenses. Use of those libraries and
+/// frameworks are governed by their own individual licenses.
+///
+/// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+/// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+/// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+/// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+/// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+/// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+/// THE SOFTWARE.
+
+import Foundation
+
+class APIStore: ObservableObject {
+  var apiFileExists = true
+  
+  struct APIS: Codable {
+    let apis: [API]
+    var count = 1425
+    
+    enum CodingKeys: String, CodingKey {
+      case apis = "entries"
+      case count
+    }
+  }
+  
+  @Published var names: APIS = APIS(apis: []) {
+    didSet {
+      saveJSON()
+    }
+  }
+  
+  let apisJSONURL = URL(fileURLWithPath: "apilist", relativeTo: FileManager.documentsDirectoryURL)
+    .appendingPathExtension("json")
+  
+  init() {
```

Comment:
Certainly! Here's an improved version of your PR comment, including a code example for clarity:

---
I noticed that `loadJSON()` isn't being called in the `onAppear` modifier within your SwiftUI view. Calling `loadJSON()` in `onAppear` ensures that your data is loaded when the view appears. Was there a specific reason for not doing this? 

Here's an example of how you could incorporate `loadJSON()` into the `onAppear` modifier:

```swift
struct ContentView: View {
    @State private var data: [YourDataType] = []

    var body: some View {
        List(data) { item in
            Text(item.name) // Customize this line based on your data structure
        }
        .onAppear {
            loadJSON()
        }
    }
```

This way, `loadJSON()` will be called when the view appears, ensuring that your data is loaded appropriately.

Reply:
Thanks, this is helpful!